src/libcore/future/future.rs

```diff
@@ -99,6 +99,21 @@ pub trait Future {
     fn poll(self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Self::Output>;
 }
 
+/// Conversion into a `Future`.
+#[unstable(feature = "into_future", issue = "0")]
+pub trait IntoFuture {
+    /// The output that the future will produce on completion.
+    #[unstable(feature = "into_future", issue = "0")]
+    type Output;
```

Comment:
Is the `IntoFuture::Output` associated type just a convenience? That is, can't we leave this out, and have people "just" write `IntoFuture::Future::Output` instead?

Reply:
It is, but its consistent with the API for IntoIterator.

Reply:
You also can't write `IntoFuture::Future::Output` - it has to be `<IntoFuture::Future as Future>::Output` D:

Reply:
Prior discussion about this: https://github.com/rust-lang/rust/pull/65244#pullrequestreview-301916859